### PR TITLE
feat(side-dag): add signers validation on peer hello

### PR DESCRIPTION
**Pull request by glevco**
Depends on https://github.com/HathorNetwork/hathor-core/pull/1075

### Motivation

This PR adds a validation of PoA signers list when peers connect.

### Acceptance Criteria

- Add consensus algorithm validation on peer hello.
- Change `Builder` so it requires settings to be set.
- Update P2P related code to use injected settings.

### Checklist

- [x] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged

### CODE REVIEW

**Review by jansegre — APPROVED** [review]

File: hathor/consensus/consensus_settings.py [hunk 1]
```diff
@@ -112,5 +130,12 @@ def _get_valid_vertex_versions(self, include_genesis: bool) -> set[TxVersion]:
 
         return versions
 
+    @override
+    def get_peer_hello_hash(self) -> str | None:
+        data = b''
+        for signer in self.signers:
```
Comment (msbrogli) [inline hathor/consensus/consensus_settings.py:136]: This code enforces the order of the signers and I think this is a good verification. I think you can safely remove the `sorted()` from the verification so we ensure that the signed id is the same for all peers and that we can check the signer id simply by checking the settings file.

File: hathor/consensus/consensus_settings.py [hunk 2]
```diff
@@ -112,5 +130,12 @@ def _get_valid_vertex_versions(self, include_genesis: bool) -> set[TxVersion]:
 
         return versions
 
+    @override
+    def get_peer_hello_hash(self) -> str | None:
```
Comment (msbrogli) [inline hathor/consensus/consensus_settings.py:134]: This is immutable. Should we calculate it once and just cache the digest?

File: hathor/consensus/consensus_settings.py (hunk 1 above)
Comment (glevco) [reply hathor/consensus/consensus_settings.py:142]: Done in https://github.com/HathorNetwork/hathor-core/pull/1078/commits/b58b125dae7da71d39f750fa53d8729f4efe3463

File: hathor/consensus/consensus_settings.py (hunk 2 above)
Comment (glevco) [reply hathor/consensus/consensus_settings.py]: Done in https://github.com/HathorNetwork/hathor-core/pull/1078/commits/b58b125dae7da71d39f750fa53d8729f4efe3463

Comment (msbrogli) [reply hathor/consensus/consensus_settings.py]: Thanks!

**Review by msbrogli — DISMISSED** [review]

**Review by msbrogli — APPROVED** [review]

**Review by msbrogli — APPROVED** [review]

**Review by jansegre — APPROVED** [review]

**Pull request merged**